### PR TITLE
docs: clean up getting started guide

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: doc/getting-started.md [hunk 1]
```diff
@@ -23,17 +23,30 @@ The configuration must be passed to Ignition through the designated data source.
...
 ```
 
 In the event that this doesn't yield any results, running as root may help. There are circumstances where the journal isn't owned by the systemd-journal group or the current user is not a part of that group.
 
-In the vast majority of cases, it will be immediately obvious why Ignition failed. If it's not, inspect the config that Ignition wrote into the log. This shows how Ignition interpreted the supplied configuration. The user-provided config may have a misspelled section or maybe an incorrect hierarchy.
+### Validating the Configuration
+
+One common cause for Ignition failures is a malformed configuration (e.g. a misspelled section or incorrect hierarchy). Ignition will log errors, warnings, and other notes about the configuration that it parsed, so this can be used to debug issues with the configuration provided. As a convenience, CoreOS hosts an [online validator][validator] which can be used to quickly verify configurations.
+
+### Enabling Services
```
Comment (dghubble) [inline doc/getting-started.md]: To the casual reader, this hops from Ignition terms to systemd terms, whereas elsewhere you say "when using systemd". Maybe title this "Systemd Services" or "Enabling Systemd Services" or "Enabling Units"

File: doc/getting-started.md [hunk 2]
```diff
@@ -23,17 +23,30 @@ The configuration must be passed to Ignition through the designated data source.
...
+
+When Ignition enables services, it doesn't directly create the symlinks necessary; it leverages [systemd presets][preset]. Presets are only evaluated on first-boot, which can result in confusion if Ignition is forced to run more than once. Any services which are enabled in the configuration after the first boot won't actually be enabled by default. `systemctl preset-all` will need to be manually invoked to enable the services.
+
+Running Ignition multiple times is not a recommended or typical use-case, so it's unlikely that this situation will arise.
```
Comment (dghubble) [inline doc/getting-started.md]: "Running multiple times is not recommended" -> "Running after first boot is not recommended"
Sometimes this comes up in the way we use Ignition on first PXE boot to install CoreOS (provision disk with CoreOS) and then use Ignition on first boot from disk (provision disk so machine assumes a role).

**Comment by dghubble** [comment]
Minor suggests, then LGTM

File: doc/getting-started.md (hunk 2 above)
Comment (crawford) [inline doc/getting-started.md]: Hmm, good point.

File: doc/getting-started.md [hunk 3]
```diff
@@ -23,17 +23,31 @@ The configuration must be passed to Ignition through the designated data source.
...
+### Validating the Configuration
+
+One common cause for Ignition failures is a malformed configuration (e.g. a misspelled section or incorrect hierarchy). Ignition will log errors, warnings, and other notes about the configuration that it parsed, so this can be used to debug issues with the configuration provided. As a convenience, CoreOS hosts an [online validator][validator] which can be used to quickly verify configurations.
+
+### Enabling systemd Services
+
+When Ignition enables systemd services, it doesn't directly create the symlinks necessary for systemd; it leverages [systemd presets][preset]. Presets are only evaluated on [first-boot][conditions], which can result in confusion if Ignition is forced to run more than once. Any systemd services which have been enabled in the configuration after the first boot won't actually be enabled after the next invocation of Ignition. `systemctl preset-all` will need to be manually invoked to create the necssary symlinks, enabling the services.
```
Comment (vcaputo) [inline doc/getting-started.md]: s/necssary/necessary/

File: doc/getting-started.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
 
-Ignition uses a JSON configuration file to represent the set of changes to be made. The format of this config is detailed [in the specification][configspec]. One of the most important parts of this config is the version number. This **must** match the version number accepted by Ignition. If the config version isn't accepted by Ignition, Ignition will fail to run and prevent the machine from booting. This can be seen by inspecting the console output of the failed instance. For more information, check out the [troubleshooting section][troubleshooting].
+Ignition uses a JSON configuration file to represent the set of changes to be made. The format of this config is detailed [in the specification][configspec] and the [MIME type][mime] registered with IANA. One of the most important parts of this config is the version number. This **must** match the version number accepted by Ignition. If the config version isn't accepted by Ignition, Ignition will fail to run and prevent the machine from booting. This can be seen by inspecting the console output of the failed instance. For more information, check out the [troubleshooting section][troubleshooting].
```
Comment (vcaputo) [inline doc/getting-started.md]: "and the MIME type registered with IANA" reads awkwardly, s/pe reg/pe has been reg/ ?

Comment (joshix) [inline doc/getting-started.md]: "MIME type is registered with IANA"

Comment (joshix) [inline doc/getting-started.md]: s/prevent the machine from booting/the machine will not boot/

Comment (joshix) [inline doc/getting-started.md]: s/failed instance/failed machine/ (consistent term)

File: doc/getting-started.md [hunk 5]
```diff
@@ -23,17 +23,31 @@ The configuration must be passed to Ignition through the designated data source.
...
+### Validating the Configuration
+
+One common cause for Ignition failures is a malformed configuration (e.g. a misspelled section or incorrect hierarchy). Ignition will log errors, warnings, and other notes about the configuration that it parsed, so this can be used to debug issues with the configuration provided. As a convenience, CoreOS hosts an [online validator][validator] which can be used to quickly verify configurations.
+
+### Enabling systemd Services
+
+When Ignition enables systemd services, it doesn't directly create the symlinks necessary for systemd; it leverages [systemd presets][preset]. Presets are only evaluated on [first-boot][conditions], which can result in confusion if Ignition is forced to run more than once. Any systemd services which have been enabled in the configuration after the first boot won't actually be enabled after the next invocation of Ignition. `systemctl preset-all` will need to be manually invoked to create the necssary symlinks, enabling the services.
+
+Forcing Ignition to run multiple times during a machine's lifetime is not a recommended or typical use-case, so it's unlikely that this situation will arise.
```
Comment (joshix) [inline doc/getting-started.md]: "Ignition is not typically run more than once during a machine's lifetime in a given role, so this situation requiring manual systemd intervention does not commonly arise."

**Comment by crawford** [comment]
Fixed up your concerns, @joshix and @vcaputo.

**Pull request merged**